Mathlib/Algebra/Group/Pi.lean

```diff
@@ -51,173 +51,20 @@ theorem Set.preimage_one {α β : Type*} [One β] (s : Set β) [Decidable ((1 :
 
 namespace Pi
 
-@[to_additive]
-instance semigroup [∀ i, Semigroup <| f i] : Semigroup (∀ i : I, f i) :=
-  { mul := (· * ·)
-    --pi_instance
-    mul_assoc := by intros; ext; exact mul_assoc _ _ _ }
-#align pi.semigroup Pi.semigroup
-#align pi.add_semigroup Pi.addSemigroup
-
-@[to_additive]
-instance commSemigroup [∀ i, CommSemigroup <| f i] : CommSemigroup (∀ i : I, f i) :=
-  { semigroup with
-    --pi_instance
-    mul_comm := by intros; ext; exact mul_comm _ _
-  }
-#align pi.comm_semigroup Pi.commSemigroup
-#align pi.add_comm_semigroup Pi.addCommSemigroup
-
-@[to_additive]
-instance mulOneClass [∀ i, MulOneClass <| f i] : MulOneClass (∀ i : I, f i) :=
-  { one := (1 : ∀ i, f i)
-    mul := (· * ·)
-    --pi_instance
-    one_mul := by intros; ext; exact one_mul _
-    mul_one := by intros; ext; exact mul_one _
-  }
-#align pi.mul_one_class Pi.mulOneClass
-#align pi.add_zero_class Pi.addZeroClass
-
-@[to_additive]
-instance invOneClass [∀ i, InvOneClass <| f i] : InvOneClass (∀ i : I, f i) :=
-  { one := (1 : ∀ i, f i)
-    inv := (· ⁻¹)
-    inv_one := by intros; ext; exact inv_one }
-
-@[to_additive]
-instance monoid [∀ i, Monoid <| f i] : Monoid (∀ i : I, f i) :=
-  { semigroup, mulOneClass with
-    npow := fun n x i => x i ^ n
-    --pi_instance
-    npow_zero := by intros; ext; exact Monoid.npow_zero _
-    npow_succ := by intros; ext; exact Monoid.npow_succ _ _
-  }
-#align pi.monoid Pi.monoid
-#align pi.add_monoid Pi.addMonoid
```

Comment:
Why did these move? I think they make sense where they were, and they certainly don't seem in scope for this PR.

Reply:
`Algebra.Group.Pi` is a bag grab of imports while `Data.Pi.Algebra` is much more tame, and the instances I moved can be defined in it rather than being postponed for no good reason.

Now in #10693